File: problemsolvingjs/medium/LengthOfLongestSubstring.js
const lengthOflongestSubstring = (s)=>{
    let right = 0;
    let left = 0;
    let maxLen = 0;
    const map = new Map();
    while(right<s.length){
        if(map.has(s.charAt(right)))
            left = Math.max(left,map.get(s.charAt(right))+1);
        map.set(s.charAt(right),right);
        maxLen = Math.max(maxLen,right-left+1)
        right++;
    }
    return maxLen
}

ans = lengthOflongestSubstring("abcabc");
console.log(ans)

File: problemsolvingpython/medium/LengthOfLongestSubstring.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        left =0
        right = 0
        maxLen =0
        n = len(s)
        map = {}

        while right<n:
            if s[right] in map:
                left = max(left,map[s[right]]+1)
            map[s[right]] = right
            maxLen = max(maxLen,(right-left)+1)
            right+=1
        return maxLen

sol = Solution()

# Call the lengthOfLongestSubstring function on the instance
result = sol.lengthOfLongestSubstring("abcabcbb")

# Print the result
print(result)